math/pi/spigotpi_test.go

```diff
@@ -0,0 +1,51 @@
+// spigotpi_test.go
+// description: Test for Spigot Algorithm for the Digits of Pi
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see spigotpi.go
+package pi
+
+import (
+	"strconv"
+	"testing"
+)
+
+var tests = []struct {
+	result string
+	n      int
+}{
+	{"314159", 6},
+	{"31415926535", 11},
+	{"314159265358", 12},
+	{"314159265358979323846", 21},
+	{"314", 3},
+	{"31415", 5},
+}
```

Comment:
Can you make this local inside the `TestPiSpigot` function or create a factory function that generates these test cases? Without a factory or local initialisation it is possible to have conflict with other implementations. I'm aware that this is the only algorithm inside this directory but if someone adds another algorithm, it could create conflict - better to be safe then sorry 😄